File: grimdb/sdk/csharp/src/Models/Workspace.cs
using System.Text.Json.Serialization;

namespace Grimlocker.Models;

public sealed record Workspace
{
    [JsonPropertyName("id")]
    public string Id { get; init; } = "";

    [JsonPropertyName("name")]
    public string Name { get; init; } = "";

    [JsonPropertyName("is_default")]
    public bool IsDefault { get; init; }
}

File: grimdb/sdk/php/src/Models/Workspace.php
<?php declare(strict_types=1);
namespace Grimlocker\Models;

class Workspace
{
    public function __construct(
        public readonly string $id,
        public readonly string $name,
        public readonly bool   $isDefault = false,
    ) {}

    public static function fromArray(array $data): self
    {
        return new self(
            $data['id']         ?? '',
            $data['name']       ?? '',
            $data['is_default'] ?? false,
        );
    }
}
